pkg/kv/kvserver/replica.go

```diff
@@ -1741,14 +1741,11 @@ func (r *Replica) isNewerThanSplit(split *roachpb.SplitTrigger) bool {
 
 func (r *Replica) isNewerThanSplitRLocked(split *roachpb.SplitTrigger) bool {
```

Comment:
Locking the mutex is no longer needed, we just read `r.replicaID` here which is immutable.